src/Microsoft.Azure.WebJobs.Host/Bindings/TraceWriter/TextWriterTraceAdapter.cs

```diff
@@ -27,14 +27,18 @@ public override Encoding Encoding
         {
             get { return Encoding.Default; }
         }
+        public static TextWriter Synchronized(TraceWriter traceWriter)
+        {
+            return TextWriter.Synchronized(new TextWriterTraceAdapter(traceWriter));
+        }
 
         public override void Write(char value)
         {
             // buffer all output
             _text.Append(value);
 
             int len = _text.Length;
-            if (len > 2 && _text[len - 2] == '\r' && _text[len - 1] == '\n')
+            if (len >= 2 && _text[len - 2] == '\r' && _text[len - 1] == '\n')
```

Comment:
In my test, I just log WriteLine(string.Empty) -- which looks like it just sends '\r\n'. In that case, the string length was only 2, so this logic didn't kick in like I would have expected. Instead, it waited until it had '\r\n\r\n' before flushing. It's a minor change -- but I thought it made sense.